internal/api/network.go

```diff
@@ -791,7 +791,7 @@ func (nm *NetworkManager) UpdateConnection(connection *Connection, name, ssid st
 	return nil
 }
 
-func (nm *NetworkManager) UpdateHotspot(hotspot *Hotspot, ssid string, password *string) error {
+func (nm *NetworkManager) UpdateHotspot(hotspot *Hotspot, ssid string, password *string, typeStr string) error {
```

Comment:
Since the type of the hotspot can never change, we do not need to pass it along the `UpdateHotspot` function.